pkg/infrastructure/alibabacloud/alibabacloud.go

```diff
@@ -0,0 +1,10 @@
+package alibabacloud
+
+import (
+	"github.com/openshift/installer/pkg/infrastructure"
+	"github.com/openshift/installer/pkg/terraform/stages/alibabacloud"
```

Comment:
This level of indirection--having a separate provider package pointing to terraform--is not helpful. Instead we should remove all of these infrastructure/provider packages (we only need them with new infra providers) and just have ProviderForPlatform point directly to terraform like in: https://github.com/openshift/installer/pull/7534/commits/32a178c4e5cd53baf602b250539e2da63aace1e8

Reply:
> Is the plan to have the infrastructure package contain the non-terraform ways to layout the infrastructure? For AWS, would it contain the cloud-sdk calls?

Yes your summary/understanding is exactly right. 

I have refactored with the change from https://github.com/openshift/installer/commit/32a178c4e5cd53baf602b250539e2da63aace1e8 so I think this should be a little less convoluted now.